src/node_crypto.cc

```diff
@@ -3174,26 +3228,55 @@ void CipherBase::SetAuthTag(const FunctionCallbackInfo<Value>& args) {
 }
 
 
-bool CipherBase::SetAAD(const char* data, unsigned int len) {
+bool CipherBase::SetAAD(const char* data, unsigned int len, int plaintext_len) {
   if (ctx_ == nullptr || !IsAuthenticatedMode())
     return false;
+
   int outlen;
-  if (!EVP_CipherUpdate(ctx_,
-                        nullptr,
-                        &outlen,
-                        reinterpret_cast<const unsigned char*>(data),
-                        len)) {
-    return false;
+  const EVP_CIPHER* const cipher = EVP_CIPHER_CTX_cipher(ctx_);
+  int mode = EVP_CIPHER_mode(cipher);
```

Comment:
`const int mode = EVP_CIPHER_CTX_mode(ctx_)`?  `cipher` does not seem to be used apart from this line.